### PR TITLE
mimic: mds: change how mds revoke stale caps

### PR DESCRIPTION
**Pull request by ukernel**
https://tracker.ceph.com/issues/38097
https://tracker.ceph.com/issues/40327

### CODE REVIEW

File: qa/tasks/cephfs/test_client_recovery.py [hunk 1]
```diff
@@ -511,11 +533,39 @@ def test_stale_renew(self):
...
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+
+        gid_str = self.mount_a.run_python(dedent("""
+            import cephfs as libcephfs
+            cephfs = libcephfs.LibCephFS(conffile='')
+            cephfs.mount()
+            client_id = cephfs.get_instance_id()
+            cephfs.abort_conn()
+            print client_id
```
Comment (batrick) [inline qa/tasks/cephfs/test_client_recovery.py]: This PR needs to backport parts of https://github.com/ceph/ceph/pull/22543

````
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:Running:
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> sudo adjust-ulimits daemon-helper kill python -c '
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> import cephfs as libcephfs
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> cephfs = libcephfs.LibCephFS(conffile='"'"''"'"')
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> cephfs.mount()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> client_id = cephfs.get_instance_id()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> cephfs.abort_conn()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> print client_id
2019-07-10T01:33:10.835 INFO:teuthology.orchestra.run.smithi191:> '
2019-07-10T01:33:11.002 INFO:teuthology.orchestra.run.smithi191.stderr:Traceback (most recent call last):
2019-07-10T01:33:11.003 INFO:teuthology.orchestra.run.smithi191.stderr:  File "<string>", line 5, in <module>
2019-07-10T01:33:11.003 INFO:teuthology.orchestra.run.smithi191.stderr:AttributeError: 'cephfs.LibCephFS' object has no attribute 'get_instance_id'
2019-07-10T01:33:11.130 INFO:teuthology.orchestra.run.smithi191.stderr:daemon-helper: command failed with exit status 1
2019-07-10T01:33:11.135 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-07-10T01:33:11.136 INFO:tasks.cephfs_test_runner:test_abort_conn (tasks.cephfs.test_client_recovery.TestClientRecovery) ... ERROR
````

From: /ceph/teuthology-archive/yuriw-2019-07-09_21:49:10-fs-wip-yuri6-testing-2019-07-09-1937mimic-testing-basic-smithi/4106965/teuthology.log

**Comment by smithfarm** [comment]
@ukernel Ping - ETA? (13.2.7 release is approaching)

**Comment by ukernel** [comment]
updated

**Comment by smithfarm** [comment]
@batrick Can you have another look?

**Comment by batrick** [comment]
good to test

**Comment by yuriw** [comment]
@ukernel pls rebase

--- pr 28585 --- pulling https://github.com/ukernel/ceph.git branch mimic-40327
remote: Enumerating objects: 101, done.
remote: Counting objects: 100% (101/101), done.
remote: Total 116 (delta 101), reused 101 (delta 101), pack-reused 15
Receiving objects: 100% (116/116), 68.90 KiB | 0 bytes/s, done.
Resolving deltas: 100% (101/101), completed with 28 local objects.
From https://github.com/ukernel/ceph
 * branch            mimic-40327 -> FETCH_HEAD
Auto-merging src/mds/SessionMap.h
Auto-merging src/mds/Server.cc
Auto-merging src/mds/Migrator.cc
Auto-merging src/mds/MDCache.h
Auto-merging src/mds/MDCache.cc
Auto-merging src/mds/Locker.h
Auto-merging src/mds/Locker.cc
CONFLICT (content): Merge conflict in src/mds/Locker.cc
Auto-merging src/common/options.cc
Auto-merging src/client/Client.h
Auto-merging src/client/Client.cc
CONFLICT (content): Merge conflict in src/client/Client.cc
Auto-merging qa/tasks/cephfs/test_misc.py
CONFLICT (content): Merge conflict in qa/tasks/cephfs/test_misc.py
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Comment by smithfarm** [comment]
@ukernel Please rebase

**Comment by smithfarm** [comment]
@batrick I dismissed your review and am re-flagging this for integration testing. Please check if that's the right thing to do.

**Comment by yuriw** [comment]
@ukernel needs rebase
--- pr 28585 --- pulling https://github.com/ukernel/ceph.git branch mimic-40327
remote: Enumerating objects: 101, done.
remote: Counting objects: 100% (101/101), done.
remote: Total 116 (delta 101), reused 101 (delta 101), pack-reused 15
Receiving objects: 100% (116/116), 25.39 KiB | 12.70 MiB/s, done.
Resolving deltas: 100% (101/101), completed with 24 local objects.
From https://github.com/ukernel/ceph
 * branch                  mimic-40327 -> FETCH_HEAD
Auto-merging src/mds/Server.cc
Auto-merging src/mds/Migrator.cc
Auto-merging src/mds/MDCache.h
Auto-merging src/mds/MDCache.cc
Auto-merging src/mds/Locker.cc
Auto-merging src/mds/CInode.cc
Auto-merging src/common/options.cc
Auto-merging src/client/Client.h
Auto-merging src/client/Client.cc
Auto-merging qa/tasks/cephfs/test_misc.py
CONFLICT (content): Merge conflict in qa/tasks/cephfs/test_misc.py
Auto-merging qa/tasks/cephfs/test_client_recovery.py
Auto-merging qa/tasks/cephfs/mount.py
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Comment by vshankar** [comment]
@ukernel rebase required -- see https://tracker.ceph.com/issues/42348

@yuriw, fyi.

**Comment by ukernel** [comment]
rebased

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-10-21-1527-mimic](https://trello.com/c/fqlT4ZTu/727-wip-yuri4-testing-2019-10-21-1527-mimic)

**Pull request merged**